src/mono/wasm/debugger/BrowserDebugProxy/MemberObjectsExplorer.cs

```diff
@@ -676,6 +695,23 @@ public GetMembersResult(JArray value, bool sortByAccessLevel)
             PrivateMembers = t.PrivateMembers;
         }
 
+        public void CleanUp()
+        {
+            CleanUpJArray(Result);
+            CleanUpJArray(PrivateMembers);
+            static void CleanUpJArray(JArray arr)
+            {
+                foreach(var item in arr)
+                {
+                    item.Children().Where(x =>
+                        x is JProperty p &&
+                        p.Name.TryConvertToProxyInternalUseProperty())
+                    .ToList()
```

Comment:
This will be allocating a new list for every `item`. Instead, we can:
```csharp
        public void CleanUp()
        {
            // avoid repeated allocations of the list
            // capacity should account for the number of
            // unique internal field names available
            List<JProperty> toRemove = new(capacity: InternalFieldName.Count);

            CleanUpJArray(Result);
            CleanUpJArray(PrivateMembers);

            void CleanUpJArray(JArray arr)
            {
                foreach (var item in arr)
                {
                    foreach (JToken jt in item.Children())
                    {
                        if (jt is JProperty p && InternalFieldName.IsKnown(p.Name))
                            toRemove.Add(p);
                    }

                    foreach (JProperty jp in toRemove)
                        jp.Remove();

                    toRemove.Clear();
                }
            }
        }
```

Reply:
Actually that list will end up growing. We can instead do:
```csharp
        public void CleanUp()
        {
            JProperty[] toRemoveInObject = new JProperty[InternalFieldName.Count];

            CleanUpJArray(Result);
            CleanUpJArray(PrivateMembers);

            void CleanUpJArray(JArray arr)
            {
                foreach (JToken item in arr)
                {
                    if (item is not JObject jobj || jobj.Count == 0)
                        continue;

                    int removeCount = 0;
                    foreach (JProperty jp in jobj.Properties())
                    {
                        if (InternalFieldName.IsKnown(jp.Name))
                            toRemoveInObject[removeCount++] = jp;
                    }

                    for (int i = 0; i < removeCount; i++)
                        toRemoveInObject[i].Remove();
                }
            }
        }
```